lib/util.js

```diff
@@ -193,3 +195,52 @@ function encode_uri(uri) {
 }
 
 exports.encode_uri = encode_uri;
+
+function fetch_rss_entries (feed_url, max_items=100, max_days_old=7) {
+    const oldest_date = moment().subtract(max_days_old, 'days');
+
+    return _fetch_rss_page(feed_url, 1, [], max_items, oldest_date);
+}
+
+function _fetch_rss_json (feed_url) {
+    return new Promise((resolve, reject) => {
+        const req = request.get(feed_url);
+        const parser = new feedparser();
+        req.pipe(parser);
```

Comment:
ah turns out I misunderstood how the event loop handles this, I think you're right. Interestingly, if an event emitter ever emits an `'error'` event and there aren't any listeners, it'll just throw an exception